rust/lance-encoding/src/encodings/physical/bitpack_fastlanes.rs

```diff
@@ -1560,3 +1564,251 @@ mod tests {
     //     check_round_trip_bitpacked(arr).await;
     // }
 }
+
+// This macro chunks the FixedWidth DataBlock, bitpacks them with 1024 values per chunk,
+// it puts the bit-width parameter in front of each chunk,
+// and the bit-width parameter has the same bit-width as the uncompressed DataBlock
+// for example, if the input DataBlock has `bits_per_value` of `16`, there will be 2 bytes(16 bits)
+// in front of each chunk storing the `bit-width` parameter.
+macro_rules! chunk_data_impl {
+    ($data:expr, $data_type:ty) => {{
+        let data_buffer = $data.data.borrow_to_typed_slice::<$data_type>();
+        let data_buffer = data_buffer.as_ref();
+
+        let bit_widths = $data
+            .get_stat(Stat::BitWidth)
+            .expect("FixedWidthDataBlock should have valid bit width statistics");
+        println!("bit_widths statistics got");
+        let bit_widths_array = bit_widths
+            .as_any()
+            .downcast_ref::<PrimitiveArray<UInt64Type>>()
+            .unwrap();
+
+        let (packed_chunk_sizes, total_size) = bit_widths_array
```

Comment:
Are `chunk_size` and `total_size` a "number of values" measurement or a "number of bytes" measurement?

Reply:
packed_chunk_sizes and total_size are "number of values" measurement, it needs to be this way because the function signature `Bitpack::unchecked_pack` requires the `output` slice type to be the same as the input slice, so when we use `total_size` to 
```
      let mut output: Vec<$data_type> = Vec::with_capacity(total_size);
```
the output size is `total_size` with number of bytes of `total_size * sizeof<each element>`